core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -570,11 +574,12 @@ final class ShuffleBlockFetcherIterator(
 
   private[storage] def throwFetchFailedException(
       blockId: BlockId,
+      mapIndex: Int,
       address: BlockManagerId,
       e: Throwable) = {
     blockId match {
-      case ShuffleBlockId(shufId, mapId, reduceId) =>
-        throw new FetchFailedException(address, shufId.toInt, mapId.toInt, reduceId, e)
+      case ShuffleBlockId(shufId, _, reduceId) =>
+        throw new FetchFailedException(address, shufId, mapIndex, reduceId, e)
```

Comment:
shall we put `mapTaskId` in `FetchFailedException`? I take a look at the code, and seems `FetchFailedException` is for logging purpose. I think it's better to see `mapTaskId` and `reduceId`, so that we can know the shuffle block id which can help debugging.

Reply:
+1

Reply:
Copy that, done in 00e78b2 and change related class and UT.